libs/api/domains/license-service/src/lib/client/firearm-license-client/firearmLicenseService.api.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  GenericLicenseClient,
+  GenericLicenseUserdataExternal,
+  GenericUserLicensePkPassStatus,
+  GenericUserLicenseStatus,
+  PkPassVerification,
+} from '../../licenceService.type'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import {
+  FirearmApplicationApi,
+  LicenseInfo,
+} from '@island.is/clients/firearm-license'
+import { User } from '@island.is/auth-nest-tools'
+import { Inject } from '@nestjs/common'
+import { parseFirearmLicensePayload } from './firearmLicenseMapper'
+import { FetchError } from '@island.is/clients/middlewares'
+
+/** Category to attach each log message to */
+const LOG_CATEGORY = 'firearmlicense-service'
+
+export class GenericFirearmLicenseApi
+  implements GenericLicenseClient<LicenseInfo> {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private firearmApi: FirearmApplicationApi,
+  ) {}
+
+  private handleError(error: Partial<FetchError>): unknown {
+    // Not throwing error if service returns 403 or 404. Log information instead.
+    if (error.status === 403 || error.status === 404) {
+      this.logger.info(`Firearm license returned ${error.status}`, {
+        exception: error,
+        message: (error as Error)?.message,
+        category: LOG_CATEGORY,
+      })
+      return null
+    }
+    this.logger.info('Firearm license fetch failed', {
```

Comment:
Shouldnt this be error or at least warning as Óli suggested?

Reply:
ups